plugins/output/flight/flight.go

```
@@ -131,11 +99,124 @@ func (f *Flight) Connect() error {
+			case float32:
+				builder.Field(i).(*array.Int64Builder).AppendValues([]int64{int64(value)}, nil)
+			case float64:
+				builder.Field(i).(*array.Int64Builder).AppendValues([]int64{int64(value)}, nil)
+			}
+		case arrow.FLOAT32:
+			metricField := getField(metric, schemaField, i)
+			switch value := metricField.(type) {
+			case int32:
+				builder.Field(i).(*array.Float32Builder).AppendValues([]float32{float32(value)}, nil)
+			case int64:
+				builder.Field(i).(*array.Float32Builder).AppendValues([]float32{float32(value)}, nil)
+			case float32:
+				builder.Field(i).(*array.Float32Builder).AppendValues([]float32{value}, nil)
+			case float64:
+				builder.Field(i).(*array.Float32Builder).AppendValues([]float32{float32(value)}, nil)
+			}
+		case arrow.FLOAT64:
+			metricField := getField(metric, schemaField, i)
+			switch value := metricField.(type) {
+			case int32:
+				builder.Field(i).(*array.Float64Builder).AppendValues([]float64{float64(value)}, nil)
+			case int64:
+				builder.Field(i).(*array.Float64Builder).AppendValues([]float64{float64(value)}, nil)
+			case float32:
+				builder.Field(i).(*array.Float64Builder).AppendValues([]float64{float64(value)}, nil)
+			case float64:
+				builder.Field(i).(*array.Float64Builder).AppendValues([]float64{value}, nil)
+			}
+		}
+	}
+}
+
+// Return the value of the metric tag with the name equal to the name of the schema field.
+func getTag(metric telegraf.Metric, schemaField arrow.Field, i int) string {
+	metricTag, wasSet := metric.GetTag(schemaField.Name)
+	if !wasSet {
+		panic(fmt.Sprintf("tag %d : %s not set", i, schemaField.Name))
+	}
+	return metricTag
+}
+
+// Return the value of the metric field with the name equal to the name of the schema field.
+func getField(metric telegraf.Metric, schemaField arrow.Field, i int) interface{} {
+	metricField, wasSet := metric.GetField(schemaField.Name)
+	if !wasSet {
+		panic(fmt.Sprintf("field %d : %s not set", i, schemaField.Name))
```

Maybe add a comment to document why `panic()` is used instead of returning an error in `Write()`. I assume it is because there is no reason for Telegraf to try and retransmit the data if a required value is missing.

---

Fixed.

---

Looks like "recuired" should be "required". If the above comment is changed to "This is to prevent Telegraf from attempting to retransmit `metric` when the required value is missing" then this comment should probably be changed as well.

---

Fixed.